### ISSUE TITLE
Intersect and surface matching (simple model)

### ISSUE THREAD

**Issue by DavidGoldwasser** [NREL/OpenStudio#2705]
After a number of users reported intersect and matching issues in SketchUp 2017 and also with the Surface Matching measure (which also does intersection) I made a simple test model with 4 spaces (offset 2 sets of two stacked spaces). Here are the steps I took.

1. Launch SketchUp 2017 and with OpenStudio 2.2.0
2. Created 4 spaces as shown below
3. Intersect in Entire Model - This appeared to do what it should do, it made one line each in the roofs of the first floor and the floors of the second story, and split each into two surfaces. The surfaces all had unique names (there had been some reports of surfaces with duplicate names, but I couldn't duplicate that)
4. Match in Entire Model - The picture below shows the results of the intersect. The two common walls were found, and one of the three common roof/floors were found, but two were missed. 

![screen shot 2017-07-25 at 9 27 21 am](https://user-images.githubusercontent.com/4857055/28682107-f9a2c7aa-72b8-11e7-8b3c-2707000a34e0.png)

View of single space showing seemingly good intersection
![screen shot 2017-07-27 at 11 12 50 am](https://user-images.githubusercontent.com/4857055/28683126-9d56275e-72bc-11e7-8dfb-a33f7c5e50d2.png)

5. I Next I quit SketchUp and launched the OpenStudio application, and loaded the model.
6. I ran "Surface Matching" measure with "intersect surfaces" set to true. There were no changes in the number of matched surfaces, which is odd, since it looks like there should be at it was a cleanly made model.
7. I Launched SketchUp and unmatched the model, and deleted the intersected geometry, then saved and quit)
8. Opened updated model back in OpenStudio and ran the surface matching measure again

This time it matched perfectly, even though under visual inspection the geometry looked the same as the SketchUp intersected geometry. I'm wondering if there is some tolerance issue that is slightly different between two two instances that is preventing it from looking like a match in the SketchUp intersected geometry vs. the OpenStudio intersected geometry. Screenshot below shows OS intersected and matched model.

![screen shot 2017-07-27 at 11 03 43 am](https://user-images.githubusercontent.com/4857055/28682889-d8942a1a-72bb-11e7-9b93-6dbcdddf0869.png)

@macumber I don't think we have to touch anything before 2.3 comes out, but I would continue to recommend that people don't even try to intersect in SketchUp for now, and use our measure for both intersection and surface matching.

One user did point out that the OpenStudio intersect makes a lot of extra surfaces to keep geometry convex. In cases of internal floors where don't care about non-convex they would like to have the ability to disable this. This is of course a feature request not a issue, but might be nice at some point. But if they have a wedding cake shape, that would still have to be sub-divided to avoid a hole.

Pre intersected test model
https://github.com/NREL/OpenStudio-files/blob/master/Git%20Issue%20Files/issue2705/box_test_pre_intersect.osm

Model after intersection and matching in SketchUp (No change if I also now intersect and match in OS app with measure)
https://github.com/NREL/OpenStudio-files/blob/master/Git%20Issue%20Files/issue2705/box_test_su_intersect.osm

Model resulting from intersection and match of clean model in OpenStudio Application using measure.
https://github.com/NREL/OpenStudio-files/blob/master/Git%20Issue%20Files/issue2705/box_test_os_intersect.osm

**Comment by meveraldo** [comment]
Hi @DavidGoldwasser 

I am working through your proposed workflow to intersect & surfacematch my model, however ran through strange result. When I run the surface matching measure in OpenStudio app, it suggests a successful outcome. After it completed the run, I saved the model and close the app.

When I then reload the model via sketchup OS plugin, the model appears exactly the same as when I saved it last: i.e. no intersection and no surface matching - as if it had never been completed. 

I am in my early days with OpenStudio and sure would appreciate some guidance from the experts. Cheers!
